components/dashboard/completedQuests.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FunctionComponent } from "react";
+import styles from "@styles/Home.module.css";
+import { QuestDocument } from "types/backTypes";
+import { useRouter } from "next/navigation";
+import UnavailableIcon from "@components/UI/iconsComponents/icons/unavailableIcon";
+import { CDNImg } from "@components/cdn/image";
+import BoostReward from "@components/quests/boostReward";
+import QuestCard from "@components/quests/questCard";
+import questCardStyles from "@styles/quests.module.css";
+
+type QuestProps = {
+  onClick: () => void;
+  imgSrc: string;
+  title: string;
+  issuer: Issuer;
+  reward: string;
+  id: number;
+  expired: boolean;
+};
+
+const QuestCardCustomised: FunctionComponent<QuestProps> = ({
+  onClick,
+  imgSrc,
+  title,
+  issuer,
+  reward,
+  id,
+  expired,
+}) => {
+  return (
+    <QuestCard
+      id={id}
+      imgSrc={imgSrc}
+      title={title}
+      issuer={issuer}
+      onClick={() => !expired && onClick()}
+      disabled={expired}
+    >
+      <div
+        className={`flex mt-2 mb-1 items-center ${
+          expired ? "opacity-40" : null
+        }`}
+      >
+        <p className="text-gray-400">{issuer.name}</p>
+      </div>
+      <div className="flex gap-2 mt-3 justify-center md:justify-start">
+        <div className={questCardStyles.issuer}>
+          {expired ? (
+            <>
+              <p className="text-white mr-2">Expired</p>
+              <UnavailableIcon width="24" color="#D32F2F" />
+            </>
+          ) : (
+            <>
+              <CDNImg width={20} src={issuer.logoFavicon} loading="lazy" />
+              <p className="text-white ml-2">{reward}</p>
+            </>
+          )}
+        </div>
+        <BoostReward questId={id} />
+      </div>
+    </QuestCard>
```

Comment:
So there are two components inside in file ? Why not makes two files ? Imo it's cleaner

Reply:
yes we don't use it anywhere else and this component heavily relies on the main component here so i add it here to make it easier to modify

Reply:
Please make two components, give this file to GPT 4 and ask him to do it he'll do it very easily :)